aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -332,10 +335,18 @@ private Completable handleIncomingUpdate() {
                     if (QueryPredicates.all().equals(incoming.getPredicate())) {
                         // If the incoming update does not have a condition, we want to delete any
                         // existing mutations for the modelId before saving the incoming one.
-                        return removeNotLocking(existing.getMutationId()).andThen(saveIncomingAndNotify());
+                        try {
+                            @SuppressWarnings("unchecked")
+                            PendingMutation<T> mergedPendingMutation = (PendingMutation<T>) merge(incoming, existing);
+                            return removeNotLocking(existing.getMutationId())
+                                    .andThen(saveIncomingAndNotify(mergedPendingMutation));
+                        } catch (Exception exception) {
```

Comment:
Generally it's best only to catch specific exceptions, rather than this top level exception.  Can you enumerate the specific ones you are expecting here instead?